packages/jsx/src/target/html/components/tabular.tsx

```diff
@@ -0,0 +1,207 @@
+import React from "react";
+import { PretextStateContext } from "../state";
+import { ReplacerComponent } from "../replacers/replacer-factory";
+import { XastElement } from "@pretext-book/jsx/src/utils/xast";
+
+const cellClasses = function (
+    tabular: XastElement,
+    r_index: number,
+    c_index: number
+): string {
+    const rows = tabular.children.filter(
+        (n): n is XastElement => n.type == "element" && n.name == "row"
+    );
+    const cols = tabular.children.filter(
+        (n): n is XastElement => n.type == "element" && n.name == "col"
+    );
+    const row: XastElement = rows[r_index];
+    const cells = row.children.filter(
+        (n): n is XastElement => n.type == "element" && n.name == "cell"
+    );
+    const cell: XastElement = cells[c_index];
+    const col: XastElement | undefined = cols[c_index];
+
+    const toNum = (s: string | null | undefined) => {
```

Comment:
We can probably tighten this up by doing it the "python" way. I.e., 

```
const TO_NUM_MAP = {major: "3", medium: "2", minor: "1"};
```

and then do
```
function toNum(s: string|null|undefined) {
    return TO_NUM_MAP[s] || "0"
}
```

Also, these functions don't change, so we can define them outside of the component so they are only created once.

If there are a bunch of these, it might be worthwhile making a `tables.ts` in the `src/utils/` directory. Then we could also write some independent tests for these functions.

Reply:
it turned out to be a bit more subtle than that (TS is unhappy unless it's convinced you're passing an actual key), but this is the approach I took in the latest commit.

I did not factor them out to a utility module however.